### DISCUSSION TITLE
Retrieving colors defined as functions, with `opacityVariable`, via `resolveConfig`

### DISCUSSION THREAD

**Discussion by simonswiss** [tailwindlabs/tailwindcss#8264] [Help]
## Defining colors as CSS variables

When creating "themable" colors by using CSS variables, the color needs to be defined as a function, to compose the internal `opacityVariable` with the RGB channles of the variable:

```css
@layer base {
  :root {
    --color-primary: 0 188 212;
  }
}
```

```diff
// tailwind.config.js
module.exports = {
  theme: {
    extend: {
      colors: {
+        primary: ({ opacityVariable }) => {
+         if (opacityVariable !== undefined) {
+            return `rgb(var(--color-primary) / var(${opacityVariable}))`
+          }
+          return 'rgb(var(--color-primary))'
+        },
      }
    }
  }
}
```

This works great to create new color utilities, but here's my problem:

## Getting colors defined as functions via `resolveConfig`

When trying to access all my theme's colors with the `resolveConfig` approach, the resolved config will give the function itself for colors defined as functions, instead of the value returned from the function.

Here's what the resolved config's color object will look like, given the Tailwind config below the screenshot:

```js
import resolveConfig from 'tailwindcss/resolveConfig'
import tailwindConfig from '../tailwind.config.js'

const fullConfig = resolveConfig(tailwindConfig)
```

<br />
<br />

<img alt="image" src="https://user-images.githubusercontent.com/485747/166608079-5889b693-1f5d-48cb-b3ef-afdbcd6c8079.png">

<br />
<br />

```js
module.exports = {
  // ...
  theme: {
    colors: {
      white: '#ffffff',
      foo: '#ff0099',
      // Colors defined as functions
      // This one won't support opacity, but it's here just to demo the resolved config output
      bar: () => '#ff9900',
      // With opacity variable
      baz: ({ opacityVariable }) => {
        if (opacityVariable !== undefined) {
          return `rgb(var(--color-primary) / var(${opacityVariable}))`
        }
        return 'rgb(var(--color-primary))'
      },
    },
  }
}
```

Is this the expected output, or was `resolveConfig` intended to give the **returned values for those colors**?

## Accessing the function values

I can obtain the color value for `bar` by calling the function:

```js
fullConfig.theme.colors.bar() => `#ff9900`
```

But... how can get the value of the function that needs the opacity variable? Can I somehow get this internal `opacityVariable` to pass it in the function?

```js
fullConfig.theme.colors.baz(????) => `rgb(0 188 212 / 1)`
```

Apologies if I'm missing a simple point here, perhaps too deep in the weeds to see the obvious 😅

**Comment by tyrauber** [comment]
Hey @simonswiss, I was wondering if you ever discovered a solution for this?  I am looking for a similar method of translating css variables with resolveConfig.
